components/derive/src/exonum_interface.rs

```diff
@@ -219,9 +271,12 @@ impl ExonumInterface {
         // Process trait methods.
         let mut methods = Vec::with_capacity(item_trait.items.len());
         let mut has_output = false;
-        let mut used_method_ids = HashSet::new();
         let mut next_method_id = 0;
 
+        // Initialize used methods with removed method IDs.
+        let mut used_method_ids: HashSet<u32> =
```

Comment:
Nit: It could make sense to have more specific error message if the existing method ID collides with one of removed IDs.